MaterialLib/MPL/Enums.h

```diff
@@ -0,0 +1,311 @@
+/**
+ * \author Norbert Grunwald
+ * \date   Sep 7, 2017
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/variant.hpp>
```

Comment:
Can we already use `std::variant`?

Reply:
Unfortunately not until gcc-7.1. And also the macOS's clang in version 10 does not support it fully yet. But (fasten your belts!!!) the latest visual studio would work!

Also tried new syntax for "structured binding declarations" instead of `std::tie`
```c++
auto [a, b, c] = function_returning_a_tuple_of_tree();
```
working only since gcc-7